src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -2204,6 +2443,381 @@ def construct_command(self, task_exec, task_args, task_numcores,
+            prop_str += '</property>\n'
+
+            prop_str += '<property>\n'
+            prop_str += '  <name>dfs.data.dir</name>\n'
+            prop_str += '    <value>file:///tmp/hadoop/hadoopdata/hdfs/datanode</value>\n'
+            prop_str += '</property>\n'
+
+            lines.insert(-1,prop_str)
+
+            hdfs_site_file = open(os.getcwd()+'/hadoop/etc/hadoop/hdfs-site.xml','w')
+            for line in lines:
+                hdfs_site_file.write(line)
+            hdfs_site_file.close()
+
+        def config_mapred_site():
+
+            mapred_site_file = open(os.getcwd()+'/hadoop/etc/hadoop/mapred-site.xml.template','r')
+            lines = mapred_site_file.readlines()
+            mapred_site_file.close()
+
+            prop_str  = ' <property>\n'
+            prop_str += '  <name>mapreduce.framework.name</name>\n'
+            prop_str += '   <value>yarn</value>\n'
+            prop_str += ' </property>\n'
+
+            lines.insert(-1,prop_str)
+
+            mapred_site_file = open(os.getcwd()+'/hadoop/etc/hadoop/mapred-site.xml','w')
+            for line in lines:
+                mapred_site_file.write(line)
+            mapred_site_file.close()
+
+        def config_yarn_site():
+
+            yarn_site_file = open(os.getcwd()+'/hadoop/etc/hadoop/yarn-site.xml','r')
+            lines = yarn_site_file.readlines()
+            yarn_site_file.close()
+
+            prop_str  = ' <property>\n'
+            prop_str += '  <name>yarn.nodemanager.aux-services</name>\n'
+            prop_str += '    <value>mapreduce_shuffle</value>\n'
+            prop_str += ' </property>\n'
+
+            lines.insert(-1,prop_str)
+
+            yarn_site_file = open(os.getcwd()+'/hadoop/etc/hadoop/yarn-site.xml','w')
+            for line in lines:
+                yarn_site_file.write(line)
+            yarn_site_file.close()
+
+        # If the LRMS used is not YARN the namenode url is going to be
+        # the first node in the list and the port is the default one, else 
+        # it is the one that the YARN LRMS returns
+        hadoop_home = None
+        if lrms.name == 'YARNLRMS':
+            logger.info('Hook called by YARN LRMS')
+            service_url    = lrms.namenode_url
+            rm_url         = "%s:%s" % (lrms.rm_ip, lrms.rm_port)
+            rm_ip          = lrms.rm_ip
+            launch_command = cls._which('yarn')
+
+        else:
+            # Here are the necessary commands to start the cluster.
+            if lrms.node_list[0] == 'localhost':
+                #Download the tar file
+                node_name = lrms.node_list[0]
+                stat = os.system("wget http://apache.claz.org/hadoop/common/hadoop-2.6.0/hadoop-2.6.0.tar.gz")
+                stat = os.system('tar xzf hadoop-2.6.0.tar.gz;mv hadoop-2.6.0 hadoop;rm -rf hadoop-2.6.0.tar.gz')
+            else:
+                node = commands.getstatusoutput('/bin/hostname')
```

Comment:
Ok, but I need the value the hostname command returns.

That applies to anywhere else I have used commands